addons/payment/models/payment_token.py

```diff
@@ -125,6 +125,19 @@ def _build_display_name(self, *args, max_length=34, should_pad=True, **kwargs):
             display_name = self.payment_details[-max_length:] if max_length > 0 else ''
         return display_name
 
+    def _get_available_tokens(self, *args, **kwargs):
+        """ Return the available tokens linked to a context defined in args/kwargs.
+
+        For a module to retrieve the linked tokens, it must override this method and add information
+        in the args/kwargs to define the context of the request.
+
+        :param tuple args: Locally unused arguments.
+        :param dict kwargs: Locally unused keywords arguments.
+        :return: The available tokens.
+        :rtype: `payment.token`
+        """
+        return self.env['payment.token']
```

Comment:
This method could be more than an empty hook. It could be used in the /payment/pay and /payment/method routes and actually return available tokens. In sale(_*) overrides, you would return something different from `super()` if the context requires it.